zonemaster/zonemaster-engine
# New "resolver.source4" and "resolver.source6" properties

## Purpose

Provide new properties "resolver.source4" and "resolver.source6" to allow configuring an IPv4 and an IPv6 source address. The "resolver.source" is untouched:

* "resolver.source" can hold an IPv4 or an IPv6 and is marked as deprecated
* "resolver.source4" can only hold an IPv4 (or the empty string) and takes precedence
* "resolver.source6" can only hold an IPv6 (or the empty string) and takes precedence

## Context

zonemaster/zonemaster-cli#315 

## Changes

* Check that if set, the "resolver.source4" property holds an IPv4 (or the empty string)
* Check that if set, the "resolver.source6" property holds an IPv6 (or the empty string)
* If set and querying over IPv4, use the "resolver.source4" value. Fallback to "resolver.source" otherwise.
* If set and querying over IPv6, use the "resolver.source6" value. Fallback to "resolver.source" otherwise.

## How to test this PR

New unit tests are provided. They should pass.

Set "resolver.source4" and/or "resolver.source6" to a value in share/profile.json and check that these interface are correctly used when running Zonemaster.
```
zonemaster-cli --show-testcase --show-module --level DEBUG --test connectivity/connectivity01 zonemaster.net
```
